test/pypi-integration.py

```diff
@@ -92,31 +93,25 @@ def test_fail_to_upload_if_input_name_different_from_package_name(self):
                 }
             )
 
-    def test_upload_when_different_names_allowed(self):
```

Comment:
This tests is wrong and broken.
- given that its correctly configured with `'name_must_match': False` option, it should be able to upload without raising `NamesValidationError`
- if we correct the above, it will race with `test_upload_pypi` test, one of which will fail because it will try to upload the same package/file twice without bumping the version.